nixos/modules/services/mail/dovecot.nix

```diff
@@ -501,6 +517,13 @@ in
         ++ optional (cfg.sieve.pipeBins != []) "sieve_extprograms";
 
       sieve.globalExtensions = optional (cfg.sieve.pipeBins != []) "vnd.dovecot.pipe";
+
+      pluginSettings = lib.mapAttrs (n: lib.mkDefault) ({
+        sieve_plugins = concatStringsSep " "cfg.sieve.plugins;
+        sieve_extensions = concatStringsSep " " (map (el: "+${el}") cfg.sieve.extensions);
+        sieve_global_extensions = concatStringsSep " " (map (el: "+${el}") cfg.sieve.globalExtensions);
```

Comment:
By setting empty extensions and plugins, subsequent definitions of extensions and plugins done in `extraConfig` won't get picked up. You're basically breaking existing configs that were completely done in `extraConfig` before.

It broke my config and I had a hard time figuring out this was caused by the previous PR. This behavior change needs at least release notes but better not render these options at all if not defined explicitly.

Reply:
You are totally right! :/

Reply:
I think what we can do is to assert for the presence of those options inside the `extraConfig` string and err out if so to let people remediate to it or not.

Reply:
What do you think about this strategy?

Reply:
That's also fine by me. Though searching through `extraConfig` is a tad error-prone since there might be comments mentioning these options. But much better than weird dovecot2 errors about missing plugins when they are clearly defined. :smiley:

Reply:
I tried something that takes into account comments, but I'm not really aware of the full grammar of Dovecot2 comments, so this is at most an OK solution.